tests/containers/docker_image.pm

```diff
@@ -30,83 +31,24 @@ sub run {
     $self->select_serial_terminal;
 
     my ($image_names, $stable_names) = get_suse_container_urls();
-    my ($plugin, $osversion, $version);
 
+    ensure_ca_certificates_suse_installed() if (is_sle());
     install_docker_when_needed();
 
-    if (is_sle && !is_public_cloud) {
-        # Allow our internal 'insecure' registry
-        assert_script_run("mkdir -p /etc/docker");
-        assert_script_run('cat /etc/docker/daemon.json; true');
-        assert_script_run(
-            'echo "{ \"insecure-registries\" : [\"registry.suse.de\", \"registry.suse.de:443\", \"registry.suse.de:5000\"] }" > /etc/docker/daemon.json');
-        assert_script_run('cat /etc/docker/daemon.json');
-        systemctl('restart docker');
-
-        if (get_var('SCC_DOCKER_IMAGE')) {
-            my $regcode = get_var 'SCC_DOCKER_IMAGE';
-            assert_script_run "cp /etc/zypp/credentials.d/SCCcredentials{,.bak}";
-            assert_script_run "echo -ne \"$regcode\" > /etc/zypp/credentials.d/SCCcredentials";
-        }
-    }
-
-    if (check_var("ARCH", "x86_64") && !is_public_cloud) {
-        zypper_call("install container-diff");
-    }
+    allow_registry_suse_de_for_docker()  if (is_sle && !is_public_cloud);
```

Comment:
`is_public_cloud` is not needed, so you can just:

```
if (is_sle) {
    ensure_ca_certificates_suse_installed();
    allow_registry_suse_de_for_docker();
}
```

Reply:
why? `is_sle` condition will be `true` in public cloud tests.

Reply:
See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10532#discussion_r442199673